### PR TITLE
Add `package_policy` fields to vulnerability and findings indexes

### PR DESCRIPTION
**Pull request by uri-weisman**
### Summary of your changes
Add `cloud_security_posture.package_policy.id` and `cloud_security_posture.package_policy.revision` to the findings and vulnerabilities indexes.

### Screenshot/Data
Screenshot 2023-09-27 at 17 44 22

### Related Issues
- https://github.com/elastic/cloudbeat/issues/1298
- https://github.com/elastic/integrations/pull/7890

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -44,6 +44,22 @@ func cloudbeatCfg(rawIn *proto.UnitExpectedConfig, agentInfo *client.AgentInfo)
...
+	}
+
+	packagePolicyRevision, ok := config["revision"].(float64)
+	if !ok {
+		return nil, fmt.Errorf("'revision' element does not exist or is not an int")
+	}
+
+	for iter := range modules {
+		modules[iter]["package_policy_id"] = packagePolicyID
+		modules[iter]["revision"] = int(packagePolicyRevision)
```
Comment (oren-zohar) [inline cmd/root.go:60]: casting to float64, the error mentions int, then casting back to an int.

Comment (oren-zohar) [inline cmd/root.go:60]: nit/opinion: `iter` here represents the index, not the value at that index. could be a little confusing given the variable name.

File: config/config.go [hunk 2]
```diff
@@ -44,14 +44,16 @@ type Fetcher struct {
...
+	Benchmark             string                  `config:"config.v1.benchmark"`
+	Type                  string                  `config:"config.v1.type"`
+	Deployment            string                  `config:"config.v1.deployment"`
+	CloudConfig           CloudConfig             `config:"config.v1"`
+	KubeConfig            string                  `config:"kube_config"`
+	Period                time.Duration           `config:"period"`
+	Processors            processors.PluginConfig `config:"processors"`
+	BundlePath            string                  `config:"bundle_path"`
+	PackagePolicyId       string                  `config:"package_policy_id"`
+	PackagePolicyRevision int                     `config:"revision"`
```
Comment (oren-zohar) [inline config/config.go:56]: any reason to have it as an int?

**Review by oren-zohar — CHANGES_REQUESTED** [review]
@amirbenun can you please take a look over the VM part?

File: config/config.go (hunk 2 above)
Comment (uri-weisman) [reply config/config.go:56]: Yes, revision is mapped as short.

File: cmd/root.go (hunk 1 above)
Comment (uri-weisman) [reply cmd/root.go]: I've removed the casting, it's not needed as we deserialize the config later on (`config.go`).

Comment (uri-weisman) [reply cmd/root.go]: changed iteration var to be `i`

File: config/config.go (hunk 2 above)
Comment (oren-zohar) [reply config/config.go:56]: is there an ECS for it? is it mapped like that throughout Elastic?

Comment (uri-weisman) [reply config/config.go:56]: could not find a related ECS schema.
I think we just followed cloud defend [mapping](https://github.com/elastic/integrations/blob/720540a54f47795a58acfa72508e995dbf028e3e/packages/cloud_defend/data_stream/file/fields/fields.yml#L8C14-L8C14)

File: config/config_test.go [hunk 3]
```diff
@@ -188,3 +188,46 @@ func (s *ConfigTestSuite) TestConfigPeriod() {
...
 }
+
+func (s *ConfigTestSuite) TestPackagePolicyFields() {
+	tests := []struct {
+		config                        string
+		expectedPackagePolicyID       string
+		expectedPackagePolicyRevision int
+	}{
+		{
+			config:                        "",
```
[the remark this answers is not in the document]
Comment (uri-weisman) [reply config/config_test.go:199]: This flow does not run as part of the `config.New` unit tests.

File: vulnerability/events_creator_test.go [hunk 4]
```diff
@@ -117,11 +119,33 @@ var combinedCvss = types.DetectedVulnerability{
...
+}
+
+func TestSuite(t *testing.T) {
+	s := new(EventsCreatorTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *EventsCreatorTestSuite) SetupTest() {
+	s.cdp = &MockGlobalDataProvider{}
+	s.bdp = &dataprovider.MockCommonDataProvider{}
```
Comment (amirbenun) [inline vulnerability/events_creator_test.go:136]: FYI, creating mocks with the constructor will also assert that all expectations met.
For example in that case you can use `NewMockGlobalDataProvider`.

**Review by amirbenun — APPROVED** [review]

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**